BlockPos.h

```diff
@@ -145,8 +145,8 @@ struct NodeCoordHashed : NodeCoord
 	NodeCoordHashed(const BlockPos &pos) : NodeCoord(pos) { rehash(); }
 	NodeCoordHashed(const NodeCoord &coord) : NodeCoord(coord) { rehash(); }
 	void rehash(void) { m_hash = NodeCoord::hash(); }
-	unsigned hash(void) { rehash(); return m_hash; }
-	unsigned hash(void) const { return m_hash; }
+	unsigned hash(void) { rehash(); return static_cast<unsigned>(m_hash); }
+	unsigned hash(void) const { return static_cast<unsigned>(m_hash); }
```

Comment:
I can't find any good reason why the `hash()` function would return an `unsigned`... Later, it is even converted back to `size_t` in `struct hash<NodeCoordHashed>`.

So it's probably better to make the `hash()` functions return size_t instead.